src/lib/party.ts

```diff
@@ -1025,21 +1017,85 @@ class Party implements Types.Party {
     let party = new Party(rawParty.body, rawParty.authorization);
     party.lazyAuthorization =
       proverParty && cloneCircuitValue(proverParty.lazyAuthorization);
-    party.children = proverParty?.children ?? [];
+    party.children = proverParty?.children ?? { parties: [] };
     party.parent = proverParty?.parent;
     return { party, result: fieldsAndResult.result };
   }
+
+  /**
+   * Like Party.witness, but lets you specify a layout for the party's children,
+   * which also get witnessed
+   */
+  static witnessTree<T>(
```

Comment:
This function is one of the main additions in this PR! It allows you to witness not just one party, but an entire tree with a specific layout